app/models/job_proxy_dispatcher.rb

```diff
@@ -90,6 +91,36 @@ def dispatch
     _log.info "Complete - Timings: #{t.inspect}"
   end
 
+  def dispatch_container_jobs
+    jobs_to_dispatch, = Benchmark.realtime_block(:pending_container_jobs) { pending_jobs(ContainerImage) }
+    Benchmark.current_realtime[:container_jobs_to_dispatch_count] = jobs_to_dispatch.length
+
+    concurrent_container_scans_limit = Settings[:container_scanning][:concurrent_per_ems].to_i
+    if jobs_to_dispatch.length > 0
+      jobs_to_dispatch.each do |job|
```

Comment:
I'd suggest you move the body of this loop into a method that dispatches a single job.